src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_BuildSynchronization.cs

```diff
@@ -3,22 +3,38 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Immutable;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
     internal partial class DiagnosticAnalyzerService
     {
+        /// <summary>
+        /// Initialize solution state for synchronizing build errors with live errors.
+        /// 
+        /// no cancellationToken since this can't be cancelled
+        /// </summary>
+        public Task InitializeSynchronizeWithBuildAsync(Solution solution, CancellationToken cancellationToken)
```

Comment:
fwiw this name reads really weird.  Maybe InitializeSynchronizationstateWithBuild instead?